controller/customdomain_utils.go

```diff
@@ -250,3 +340,101 @@ func GetInfrastructureObject(kclient client.Client) (*configv1.Infrastructure, e
 
 	return infrastructure, nil
 }
+
+// Taken from https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/utils/clusterversion.go
+// GetClusterVersionObject returns the canonical ClusterVersion object
+// To check current version: `output.Status.History[0].Version`
+//
+// `history contains a list of the most recent versions applied to the cluster.
+// This value may be empty during cluster startup, and then will be updated when a new update is being applied.
+// The newest update is first in the list and it is ordered by recency`
+//
+// Note:
+// This can be queried inside the controllers, caching doesn't apply if the scope is global rather than namespaced.
+func GetClusterVersionObject(kclient client.Client) (*configv1.ClusterVersion, error) {
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "",
+		Version: "config.openshift.io/v1",
+		Kind:    "ClusterVersion",
+	})
+	ns := types.NamespacedName{
+		Namespace: "",
+		Name:      "version",
+	}
+	err := kclient.Get(context.TODO(), ns, u)
+	if err != nil {
+		return nil, err
+	}
+
+	uContent := u.UnstructuredContent()
+	var cv *configv1.ClusterVersion
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(uContent, &cv)
+	if err != nil {
+		return nil, err
+	}
+
+	return cv, nil
+}
+
+// Taken from https://github.com/openshift/cloud-ingress-operator/blob/master/pkg/utils/clusterversion.go
+// SetClusterVersion sets the cluster version globally(to ENV as CLUSTER_VERSION)
+func SetClusterVersion(kclient client.Client) error {
+	versionObject, err := GetClusterVersionObject(kclient)
+	if err != nil {
+		return err
+	}
+
+	// handle when there's no object defined || no version found on history
+	if len(versionObject.Status.History) == 0 || versionObject == nil {
+		return fmt.Errorf("version couldn't be grabbed from clusterversion: %+v", versionObject) // (%+v) adds field names
+	}
+
+	return os.Setenv("CLUSTER_VERSION", versionObject.Status.History[0].Version)
+}
+
+func isUsingNewManagedIngressFeature(kclient client.Client, reqLogger logr.Logger) (bool, error) {
+	reqLogger.Info("Fetching cluster namespace with labels")
```

Comment:
"cluster namespace" doesn't really make sense, we're fetching from the namespace openshift-custom-domains-operator

```suggestion
	reqLogger.Info("Fetching labels from namespace", "namespace", config.OperatorNamespace)
```